formats/src/main/scala/org/scanamo/DerivedDynamoFormat.scala

```diff
@@ -35,75 +33,74 @@ trait DerivedDynamoFormat {
     fieldWitness: Witness.Aux[K]
   ): ValidConstructedDynamoFormat[FieldType[K, V] :: T] =
     new ValidConstructedDynamoFormat[FieldType[K, V] :: T] {
-      def read(av: AttributeValue): Validated[InvalidPropertiesError, FieldType[K, V] :: T] = {
-        val fieldName = fieldWitness.value.name
-
-        val possibleValue =
-          av.getM.asScala.get(fieldName).map(headFormat.value.read).orElse(headFormat.value.default.map(Either.right))
+      private final val fieldName = fieldWitness.value.name
 
-        val valueOrError = possibleValue.getOrElse(Either.left[DynamoReadError, V](MissingProperty))
+      final def read(av: DynamoObject) = {
+        val valueOrError: Either[(String, DynamoReadError), V] =
+          av(fieldName).map(headFormat.value.read) orElse
+            headFormat.value.default.map(Either.right) getOrElse
+            Either.left(MissingProperty) leftMap
+            ((fieldName, _))
 
-        def validatedProperty(x: Either[DynamoReadError, V]): Validated[InvalidPropertiesError, V] =
-          x.leftMap(e => InvalidPropertiesError(NonEmptyList(PropertyReadError(fieldName, e), Nil))).toValidated
-
-        val head: Validated[InvalidPropertiesError, FieldType[K, V]] = validatedProperty(valueOrError).map(field[K](_))
+        val head = valueOrError.toValidatedNel.map(field[K](_))
         val tail = tailFormat.value.read(av)
 
         cats.Apply[ValidatedPropertiesError].map2(head, tail)(_ :: _)
       }
-      def write(t: FieldType[K, V] :: T): AttributeValue = {
+
+      final def write(t: FieldType[K, V] :: T) = {
         val tailValue = tailFormat.value.write(t.tail)
         val av = headFormat.value.write(t.head)
-        if (!(av.isNULL eq null) && av.isNULL)
+        if (av.isNull)
```

Comment:
You removed the first part of the condition. Was it unnecessary?

Reply:
`av : DynamoValue` which [fixes the behaviour](https://github.com/scanamo/scanamo/pull/400/files#diff-d84ea639301de24601ffe448fea7a8dcR179) of null handling in `AttributeValue`.